RecoParticleFlow/PFProducer/interface/TableDefinitions.h

```diff
@@ -0,0 +1,191 @@
+
+template <class C>
+edm::ElementID refToElementID(const edm::Ref<C>& ref) {
+  return edm::ElementID(ref.id(), ref.index());
+}
+
+template <class C>
+edm::ElementID refToElementID(const edm::RefToBase<C>& ref) {
+  return edm::ElementID(ref.id(), ref.key());
+}
+
+namespace edm::soa {
+  namespace col::pf {
+    namespace track {
+      SOA_DECLARE_COLUMN(Pt, double, "Pt");
+      SOA_DECLARE_COLUMN(ExtrapolationValid, bool, "ExtrapolationValid");
+      SOA_DECLARE_COLUMN(Eta, double, "Eta");
+      SOA_DECLARE_COLUMN(Phi, double, "Phi");
+      SOA_DECLARE_COLUMN(Posx, double, "Posx");
+      SOA_DECLARE_COLUMN(Posy, double, "Posy");
+      SOA_DECLARE_COLUMN(Posz, double, "Posz");
+      SOA_DECLARE_COLUMN(PosR, double, "PosR");
+      SOA_DECLARE_COLUMN(ConvBremRefKey, edm::ElementID, "ConvBremRefKey");
+      SOA_DECLARE_COLUMN(ConvBremRefBaseKey, edm::ElementID, "ConvBremRefBaseKey");
+      SOA_DECLARE_COLUMN(IsLinkedToDisplacedVertex, bool, "IsLinkedToDisplacedVertex");
+      SOA_DECLARE_COLUMN(GsfTrackRefPFKey, edm::ElementID, "GsfTrackRefPFKey");
+      SOA_DECLARE_COLUMN(GsfTrackRefPFIsNonNull, bool, "GsfTrackRefPFIsNonNull");
+      SOA_DECLARE_COLUMN(KfPFRecTrackRefKey, edm::ElementID, "KfPFRecTrackRefKey");
+      SOA_DECLARE_COLUMN(KfPFRecTrackRefIsNonNull, bool, "KfPFRecTrackRefIsNonNull");
+      SOA_DECLARE_COLUMN(KfTrackRefKey, edm::ElementID, "KfTrackRefKey");
+      SOA_DECLARE_COLUMN(KfTrackRefBaseKey, edm::ElementID, "KfTrackRefBaseKey");
+      SOA_DECLARE_COLUMN(KfTrackRefIsNonNull, bool, "KfTrackRefIsNonNull");
+      SOA_DECLARE_COLUMN(ConvRefKey, edm::ElementID, "ConvRefKey");
+      SOA_DECLARE_COLUMN(ConvRefIsNonNull, bool, "ConvRefIsNonNull");
+      SOA_DECLARE_COLUMN(V0RefKey, edm::ElementID, "V0RefKey");
+      SOA_DECLARE_COLUMN(V0RefIsNonNull, bool, "V0RefIsNonNull");
+      SOA_DECLARE_COLUMN(TrackType_FROM_GAMMACONV, bool, "TrackType_FROM_GAMMACONV");
+      SOA_DECLARE_COLUMN(TrackType_FROM_V0, bool, "TrackType_FROM_V0");
+      SOA_DECLARE_COLUMN(GsfTrackRefPFTrackId, int, "GsfTrackRefPFTrackId");
+      SOA_DECLARE_COLUMN(DisplacedVertexRef_TO_DISP_IsNonNull, bool, "DisplacedVertexRef_TO_DISP_IsNonNull");
+      SOA_DECLARE_COLUMN(DisplacedVertexRef_TO_DISP_Key, edm::ElementID, "DisplacedVertexRef_TO_DISP_Key");
+      SOA_DECLARE_COLUMN(DisplacedVertexRef_FROM_DISP_IsNonNull, bool, "DisplacedVertexRef_FROM_DISP_IsNonNull");
+      SOA_DECLARE_COLUMN(DisplacedVertexRef_FROM_DISP_Key, edm::ElementID, "DisplacedVertexRef_FROM_DISP_Key");
+    };  // namespace track
+    namespace rechit {
+      SOA_DECLARE_COLUMN(DetIdValue, unsigned int, "DetIdValue");
+      SOA_DECLARE_COLUMN(Fraction, double, "Fraction");
+      SOA_DECLARE_COLUMN(Eta, double, "Eta");
+      SOA_DECLARE_COLUMN(Phi, double, "Phi");
+      SOA_DECLARE_COLUMN(Posx, double, "Posx");
+      SOA_DECLARE_COLUMN(Posy, double, "Posy");
+      SOA_DECLARE_COLUMN(Posz, double, "Posz");
+
+      SOA_DECLARE_COLUMN(Corner0x, float, "Corner0x");
```

Comment:
it may be useful to add D to the double-precision cases to reduce confusion with float/double mix

Reply:
Here as well the purpose of mixed types was to not have even minor reco differences, we keep them exactly as in the original. The plan was to follow up with cleaning types in a separate PR, but this could also be done here.